packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -768,10 +768,14 @@ class ModalBottomSheetRoute<T> extends PopupRoute<T> {
   /// {@macro flutter.widgets.DisplayFeatureSubScreen.anchorPoint}
   final Offset? anchorPoint;
 
-  /// If useSafeArea is true, a [SafeArea] is inserted.
+  /// Specifies the sheet's relationship to the safe area.
   ///
-  /// If useSafeArea is false, the bottom sheet is aligned to the bottom of the page
-  /// and isn't exposed to the top padding of the MediaQuery.
+  /// If true, a [SafeArea] with `bottom: false` is inserted, so the sheet
+  /// doesn't extend into the top or horizontal insets but still extends through
+  /// the bottom inset.
+  ///
+  /// If false, a [SafeArea] is not inserted, and the content is un-exposed to
+  /// the top padding via `MediaQuery.removePadding` with `removeTop: true`.
```

Comment:
Is there a way to word this that provides more of a high-level story for what this field means?

In the existing version, I feel like I have that for the "true" case, but not for the "false" case. I don't really understand what "exposed to the top padding" or "un-exposed" etc. means; as best I understand what [`MediaQuery.removePadding`](https://api.flutter.dev/flutter/widgets/MediaQuery/MediaQuery.removePadding.html) actually does, I'm not seeing how "exposed" is the right verb for it.

Modulo that, I think this could look something like:
```dart
  /// Whether to avoid system intrusions on the top, left, and right.
  ///
  /// If true, a [SafeArea] is used in order to keep the bottom sheet away from
  /// system intrusions at the top, left, and right sides of the screen.
  ///
  /// If false, [??? I don't really understand what the purpose of the `removeTop: true`
  /// is, so I'm not sure how to write this paragraph.]
  ///
  /// In either case, the bottom sheet extends all the way to the bottom of
  /// the screen, including any system intrusions.
  ///
  /// The default is false.
```

Some of the phrasing there, like "avoid system intrusions", is borrowed from the `SafeArea` docs.

---

A related question: in what sort of situation would a developer want to set this to true, or to leave it at its default of false?

Reply:
> [??? I don't really understand what the purpose of the `removeTop: true` is, so I'm not sure how to write this paragraph.]

I'm not sure about this either. 🤔 The PR that added `useSafeArea` is #107140, and its description links to https://github.com/flutter/flutter/issues/39205#issuecomment-726245495 as the motivation. That comment describes a use case I hadn't considered:

> In this case, the scrollable ModalBottomSheet has a pinned AppBar at the top, and so expected to fill the status bar space.

Then in a comment shortly after (https://github.com/flutter/flutter/issues/39205#issuecomment-726257303):

> It's a really interesting use case! The bottomsheet actually expands as it scrolls, and then the sheet/appbar kind of 'docks' with the top of the screen when it reaches it.

For that use case, I see why one might not want `ModalBottomSheetRoute` to add its own `SafeArea`: that `SafeArea` could prevent the sheet, and so its built-in app bar, from "docking" / extending all the way to the top of the screen, through the inset / system intrusion there.

But the `MediaQuery.removePadding` with `removeTop` isn't necessary to accomplish that, and in fact I think it could be unhelpful: the sheet extends up through the top inset, but the sheet's interactive content still needs to go below the top inset. An app bar normally handles that by extending its surface upward through the top inset, but with that `MediaQuery.removePadding`, it can only guess at how far that extension should go since we'll have deprived it of the `MediaQueryData.padding.top`.